smithy-swift-codegen/src/test/kotlin/StructDecodeGenerationTests.kt

```diff
@@ -91,23 +90,26 @@ class StructDecodeGenerationTests {
                     var encodeContainer = encoder.container(keyedBy: CodingKeys.self)
                     if let intList = intList {
                         var intListContainer = encodeContainer.nestedUnkeyedContainer(forKey: .intList)
-                        for intlist0 in intList {
-                            try intListContainer.encode(intlist0)
+                        for integer0 in intList {
```

Comment:
if there is a list inside this, would `integer0` name collide?

Similarly for others. We must make sure names are unique.

Reply:
If there is another `integer` it would be suffixed with `1`, `2`, `3`, etc. so name collisions would be prevented.